homeassistant/components/prosegur/config_flow.py

```diff
@@ -0,0 +1,77 @@
+"""Config flow for Prosegur Alarm integration."""
+import logging
+
+from pyprosegur.auth import COUNTRY, Auth
+from pyprosegur.installation import Installation
+import voluptuous as vol
+
+from homeassistant import config_entries, core, exceptions
+from homeassistant.helpers import aiohttp_client
+
+from .const import DOMAIN  # pylint:disable=unused-import
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {"username": str, "password": str, "country": vol.In(COUNTRY.keys())}
+)
+
+
+async def validate_input(hass: core.HomeAssistant, data):
+    """Validate the user input allows us to connect."""
+    try:
+        session = aiohttp_client.async_get_clientsession(hass)
+        auth = Auth(session, data["username"], data["password"], data["country"])
+        install = await Installation.retrieve(auth)
+    except ConnectionRefusedError:
+        raise InvalidAuth from ConnectionRefusedError
+
+    # Return info that you want to store in the config entry.
+    return {
+        "title": f"Contract {install.contract}",
+        "contract": install.contract,
+        "username": data["username"],
+        "password": data["password"],
+        "country": data["country"],
+    }
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Prosegur Alarm."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_CLOUD_POLL
+
+    async def async_step_user(self, user_input=None):
+        """Handle the initial step."""
+        errors = {}
+
+        if user_input is None:
```

Comment:
If you change this to `if user_input:`, you don't need the duplicate `self.async_show_form` in your code. Just wrap your validate input in this if.